src/main/resources/templates/project-commit-graph.ftl

```diff
@@ -0,0 +1,129 @@
+[#macro listTags repository commitId]
+    [#list repository.getTags() as tag]
+        [#if tag.getCommit().getCommit() == commitId]
+        <span class="label label-primary">${tag.getSimpleName()}</span>
+        [/#if]
+    [/#list]
+[/#macro]
+
+[#import "macros.ftl" as macros]
+[#import "components/project-frameset.ftl" as projectFrameset]
+[#import "components/commit-row.ftl" as commitRow]
+
+[@macros.renderHeader i18n.translate("section.projects") /]
+[@macros.renderMenu i18n user /]
+<div class="container">
+
+[#if repositoryEntity?? && repositoryEntity?has_content]
+    [@projectFrameset.renderBreadcrumb i18n group![] repositoryEntity/]
+[/#if]
+
+    <div class="row">
+        <div class="col-md-10 col-md-offset-2">
+            <h4 style="line-height:34px; margin-top:0;">Nice graphs &nbsp; 😍</h4>
+        </div>
+    </div>
+
+    <div class="row">
+        <div class="col-md-2">
+        [@projectFrameset.renderSidemenu "insights" i18n group![] repository/]
+        </div>
+
+        <!-- Graph for all commits -->
+        <div class="col-md-10">
+            <div id="allcommits_div" style="width: 100%; height: 500px; "></div>
+        </div>
+    </div>
+
+    <!-- Graph for a person for a commit -->
+    <div class="row" id="personcommit_divs">
+    </div>
+
+</div> <!-- closes div class="container" -->
+
+[@macros.renderScripts /]
+<script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+<script type="text/javascript" src="http://code.stephenmorley.org/javascript/colour-handling-and-processing/Colour.js"></script>
+<script type="text/javascript">
+
+    function getRandColor(brightness){
```

Comment:
Did you copy this from an external resource? Would be better to include a link to it, maybe StackOverflow?